subprojects/packagefiles/libjpeg-turbo/simd/meson.build

```diff
@@ -58,6 +69,190 @@ if host_cpu in ['x86', 'x86_64']
+    cdata_neon.set('HAVE_VLD1Q_U8_X4', cc.compiles('''
+      #include <arm_neon.h>
+      int main(int argc, char **argv) {
+        uint8_t input[] = {
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc,
+          (uint8_t)argc, (uint8_t)argc, (uint8_t)argc, (uint8_t)argc
+        };
+        uint8x16x4_t output = vld1q_u8_x4(input);
+        vst4q_u8(input, output);
+        return (int)input[0];
+      }''', args: neon_flags, name: 'supports vld1q_u8_x4 intrinsic'))
+
+    subdir('arm')
+    simd_src_arm += [neon_compat_h]
+
+    # GCC 11 and earlier and some older versions of Clang do not have a full or
+    # optimal set of Neon intrinsics, so for performance reasons, when using those
+    # compilers, we default to using the older GAS implementation of the Neon SIMD
+    # extensions for certain algorithms.  The presence or absence of the three
+    # intrinsics we tested above is a reasonable proxy for this, except with GCC 10
+    # and 11.
+
+    default_neon_intrinsics = cdata_neon.get('HAVE_VLD1_S16_X3') and cdata_neon.get('HAVE_VLD1_U16_X2') and cdata_neon.get('HAVE_VLD1Q_U8_X4') and (cc.get_id() != 'gcc' or cc.version().version_compare('>= 12.0.0'))
+
+    neon_intrinsics = get_option('neon-intrinsics').disable_auto_if(not  default_neon_intrinsics).enabled()
+
+    if not neon_intrinsics
+      if (host_cpu == 'armv7')
+        gastest = '''
+        .text
+        .fpu neon
+        .arch armv7a
+        .object_arch armv4
+        .arm
+        pld [r0]
+        vmovn.u16 d0, q0
+        '''
+      else
+        gastest = '''
+        .text
+        MYVAR .req x0
+        movi v0.16b, #100
+        mov MYVAR, #100
+        .unreq MYVAR
+        '''
+      endif
+      f = configure_file(
+        command: [python, '-c', 'import sys; print(sys.argv[1])', '@0@'.format(gastest)],
+        input: 'CMakeLists.txt',
+        output: 'gastest.S',
+        capture: true,
+      )
```

Comment:
That `input: 'CMakeLists.txt',` argument looks like a mistake.

Reply:
No, this is intended. The `command` is what echoes the real contents of `gastest.S`. `configure_file`'s `input` argument is still required, so in order to generate the test file, I need to supply it.

Reply:
The documentation is confusing. It seems to work fine without it (tested with meson 0.64, 1.2.0 & 1.5.1).

A comment explaining why there's a need to go through a generated file would be great (because the command line used by meson for `cc.compile` put the source file before the extra arguments, so `['-x', 'assembler-with-cpp']` has no impact). Which also means that those arguments can be removed, no?